packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -129,7 +132,11 @@ export default function BlockToolbar( { hideDragHandle } ) {
 								onDragStart={ onDraggableStart }
 								onDragEnd={ onDraggableEnd }
 							>
-								<BlockSwitcher clientIds={ blockClientIds } />
+								{ shouldShowBlockSwitcher && (
```

Comment:
I wonder if this setting is needed, the switcher is only unhelpful on the navigation block which doesn't seem to have a toolbar anyway due to the `! isRootBlockSelected` condition in this PR (in packages/edit-navigation/src/components/navigation-editor/block-editor-area.js). 

For navigation links it acts as the block icon but doesn't do much and isn't harmful.

I'd personally lean towards making this PR as small as possible and dropping this.

If there are parts of the switcher menu that are not useful (e.g. block styles), it might be better to add a way to disable those features for edit-navigation.